pkgs/development/compilers/factor-lang/unwrapped.nix

```
@@ -0,0 +1,107 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, curl
+, git
+, ncurses
+, tzdata
+, unzip
+}:
+let
+  rev = "e10b64dbc53a8583098e73580a1eb9ff4ce0c709";
+
+in
+stdenv.mkDerivation (finalArgs: {
+  pname = "factor-lang";
+  version = "0.99";
+
+  src = fetchurl {
+    url =
+      "https://downloads.factorcode.org/releases/${finalArgs.version}/factor-src-${finalArgs.version}.zip";
+    sha256 = "f5626bb3119bd77de9ac3392fdbe188bffc26557fab3ea34f7ca21e372a8443e";
+  };
+
+  patches = [
+    # Use full path to image while bootstrapping
+    ./staging-command-line-0.99-pre.patch
+    # Point work vocabulary root to a writable location
+    ./workdir-0.99-pre.patch
+    # Patch hard-coded FHS paths
+    ./adjust-paths-in-unit-tests.patch
+    # Avoid using /sbin/ldconfig
+    ./ld.so.cache-from-env.patch
+  ];
+
+  nativeBuildInputs = [ git makeWrapper curl unzip ];
+
+  postPatch = ''
+    sed -ie '4i GIT_LABEL = heads/master-${finalArgs.passthru.rev}' GNUmakefile
+
+    # Some other hard-coded paths to fix:
+    sed -i 's#/usr/share/zoneinfo/#${tzdata}/share/zoneinfo/#g' \
+      extra/tzinfo/tzinfo.factor
+
+    sed -i 's#/usr/share/terminfo#${ncurses.out}/share/terminfo#g' \
+      extra/terminfo/terminfo.factor
+
+    # update default paths in factor-listener.el for fuel mode
+    substituteInPlace misc/fuel/fuel-listener.el \
+      --replace '(defcustom fuel-factor-root-dir nil' "(defcustom fuel-factor-root-dir \"$out/lib/factor\""
+  '';
+
+  dontConfigure = true;
+
+  buildPhase = ''
+    runHook preBuild
+    patchShebangs ./build.sh
+    # Factor uses XDG_CACHE_HOME for cache during compilation.
+    # We can't have that. So, set it to $TMPDIR/.cache
+    export XDG_CACHE_HOME=$TMPDIR/.cache && mkdir -p $XDG_CACHE_HOME
+
+    make -j$NIX_BUILD_CORES linux-x86-64
+    printf "First build from upstream boot image\n" >&2
+    ./build.sh bootstrap
+    printf "Rebuild boot image\n" >&2
+    ./factor -script -e='"unix-x86.64" USING: system bootstrap.image memory ; make-image save 0 exit'
+    printf "Second build from local boot image\n" >&2
+    ./build.sh bootstrap
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/lib/factor
+    cp -r factor factor.image libfactor.a libfactor-ffi-test.so \
+        LICENSE.txt README.md basis core extra misc $out/lib/factor
+
+    # install fuel mode for emacs
+    mkdir -p $out/share/emacs/site-lisp
+    (cd $out/share/emacs/site-lisp; ln -s ../../../lib/factor/misc/fuel/*.el .)
```

```suggestion
    ln -s $out/lib/factor/misc/fuel/*.el $out/share/emacs/site-lisp/
```
that should be the same

---

AFAIK, the fixup phase repairs those links, making them relative again. I took this as the hint to not link them to absolute paths.

---

That doesn't really matter but you can give ln -r to make it relative immediately